db/change_cache_test.go

```diff
@@ -2300,3 +2300,723 @@ func TestSkippedSequenceCompact(t *testing.T) {
+		TimeSaved:    time.Now(),
+	}
+	_ = testChangeCache.processEntry(ctx, entry)
+
+	entry = &LogEntry{
+		Sequence:     18,
+		DocID:        fmt.Sprintf("doc_%d", 50),
+		RevID:        "1-abcdefabcdefabcdef",
+		TimeReceived: time.Now(),
+		TimeSaved:    time.Now(),
+	}
+	_ = testChangeCache.processEntry(ctx, entry)
+
+	// process unusedSeq range with pending range containing duplicate sequences that are in pending list
+	// Pending should contain: (10-13), 14, (15-17) 18, (19-20)
+	testChangeCache.releaseUnusedSequenceRange(ctx, 10, 20, time.Now())
+
+	// assert pending has 4 elements
+	require.EventuallyWithT(t, func(c *assert.CollectT) {
+		testChangeCache.updateStats(ctx)
+		assert.Equal(c, int64(5), dbContext.DbStats.CacheStats.PendingSeqLen.Value())
+		assert.Equal(c, int64(0), dbContext.DbStats.CacheStats.SkippedSeqLen.Value())
+		assert.Equal(c, int64(0), dbContext.DbStats.CacheStats.NumCurrentSeqsSkipped.Value())
+		assert.Equal(c, uint64(1), testChangeCache.nextSequence)
+		dbContext.UpdateCalculatedStats(ctx)
+		assert.Equal(c, int64(20), dbContext.DbStats.CacheStats.HighSeqCached.Value())
+	}, time.Second*10, time.Millisecond*100)
+
+	// unblock pending and assert items are processed correct
+	testChangeCache.releaseUnusedSequenceRange(ctx, 1, 10, time.Now())
+
+	// assert pending empties
+	require.EventuallyWithT(t, func(c *assert.CollectT) {
+		testChangeCache.updateStats(ctx)
+		assert.Equal(c, int64(0), dbContext.DbStats.CacheStats.PendingSeqLen.Value())
+		assert.Equal(c, int64(0), dbContext.DbStats.CacheStats.SkippedSeqLen.Value())
+		assert.Equal(c, int64(0), dbContext.DbStats.CacheStats.NumCurrentSeqsSkipped.Value())
+		assert.Equal(c, uint64(21), testChangeCache.nextSequence)
+		dbContext.UpdateCalculatedStats(ctx)
+		assert.Equal(c, int64(20), dbContext.DbStats.CacheStats.HighSeqCached.Value())
+	}, time.Second*10, time.Millisecond*100)
+}
+
+// TestReleasedSequenceRangeHandlingDuplicateSequencesInSkipped:
+//   - Test releasing unused sequence range that has duplicate sequences in skipped sequence list
+//   - Assert sequences in the range are removed from skipped
+//   - Empty skipped through unused sequence release
+//   - Add new contiguous sequence and assert it is processed correctly
+func TestReleasedSequenceRangeHandlingDuplicateSequencesInSkipped(t *testing.T) {
+	base.SetUpTestLogging(t, base.LevelDebug, base.KeyCache)
+
+	ctx := base.TestCtx(t)
+	bucket := base.GetTestBucket(t)
+	dbContext, err := NewDatabaseContext(ctx, "db", bucket, false, DatabaseContextOptions{
+		Scopes: GetScopesOptions(t, bucket, 1),
+	})
+	require.NoError(t, err)
+	defer dbContext.Close(ctx)
+
+	err = dbContext.StartOnlineProcesses(ctx)
+	require.NoError(t, err)
+
+	ctx = dbContext.AddDatabaseLogContext(ctx)
+	testChangeCache := &changeCache{}
+	if err := testChangeCache.Init(ctx, dbContext, dbContext.channelCache, nil, &CacheOptions{
+		CachePendingSeqMaxWait: 2 * time.Minute,
+		CacheSkippedSeqMaxWait: 2 * time.Minute,
```

Comment:
If the test runs very slow (> 2 minute) for some reason, these values will break some of the assertions below (as things will start being ejected from pending and skipped.  Is there a reason to not bump these up to a larger value?

Reply:
Upped the time for all 2 minute waits to 20 minutes.